iceoryx2-cli/iox2-service/src/cli.rs

```diff
@@ -142,6 +143,156 @@ pub struct ListenOptions {
     pub repetitions: Option<u64>,
 }
 
+#[derive(Clone, Copy, ValueEnum)]
+#[value(rename_all = "UPPERCASE")]
+pub enum CliTypeVariant {
+    Dynamic,
+    FixedSize,
+}
+
+#[derive(Clone, Copy, ValueEnum)]
+#[value(rename_all = "UPPERCASE")]
+pub enum DataRepresentation {
+    Iox2Dump,
+    Hex,
+}
+
+#[derive(Parser)]
+pub struct PublishOptions {
+    #[clap(help = "Name of the service which shall the message be sent to.")]
+    pub service: String,
+    #[clap(
+        short,
+        long,
+        default_value = "shell_node",
+        help = "Defines the node name of the publish endpoint."
+    )]
+    pub node_name: String,
+    #[clap(
+        short,
+        long,
+        help = "The messages that shall be sent. Can be multiple messages. If no messages are given stdin is read."
+    )]
+    pub message: Vec<String>,
+    #[clap(short, long, help = "When set, the data from this file is published.")]
+    pub input_file: Option<String>,
+    #[clap(
+        short,
+        long,
+        default_value = "1000",
+        help = "Time between the messages in milliseconds."
+    )]
+    pub time_between_messages: usize,
+
+    #[clap(
+        short,
+        long,
+        default_value = "HEX",
+        help = "Defines how the provided data is encoded."
+    )]
+    pub data_representation: DataRepresentation,
+
+    #[clap(
+        short,
+        long,
+        default_value = "1",
+        help = "How often shall the messages be sent."
+    )]
+    pub repetitions: usize,
```

Comment:
I think it should be not too difficult to have something like this

```rs
enum Repetitions {
    Count(u64),
    Infinite,
}

Reply:
No, please not. Then leave it rather as it is. Surrogates are a bad idea and should not be use.

Reply:
I know you will hate me for this comment, but please remove this. Using surrogates is an anti feature and makes it harder to solve the problem later in a proper way.